src/platform/ESP32/DiagnosticDataProviderImpl.cpp

```diff
@@ -287,6 +287,61 @@ void DiagnosticDataProviderImpl::ReleaseNetworkInterfaces(NetworkInterface * net
     }
 }
 
+CHIP_ERROR DiagnosticDataProviderImpl::GetThreadMetrics(ThreadMetrics ** threadMetricsOut)
+{
+#ifdef CONFIG_FREERTOS_USE_TRACE_FACILITY
+    TaskStatus_t * taskStatusArray = nullptr;
+    ThreadMetrics * head           = nullptr;
+    uint32_t arraySize             = uxTaskGetNumberOfTasks();
+
+    taskStatusArray = (TaskStatus_t *) chip::Platform::MemoryAlloc(arraySize * sizeof(TaskStatus_t));
```

Comment:
```suggestion
    auto taskStatusArray = static_cast<TaskStatus_t *>(Platform::MemoryAlloc(arraySize * sizeof(TaskStatus_t)));
```
and remove the declaration above; declare things where they are actually needed.